test/pageobjects/home.page.js

```diff
@@ -0,0 +1,58 @@
+import Page from './page.js';
+
+class HomePage extends Page {
+    /**
+     * define selectors using getter methods
+     */
+    get cartIcon(){
+        return $('[data-test="shopping-cart-link"]');
+    }
+
+    get productSortContainer(){
+        return $('[data-test="product-sort-container"]');
+    }
+
+    // Custom locator function for product name using XPath
+    getProductByName(productName) {
+        return `//*[contains(text(), "${productName}")]`;
+    }
+
+    async clickProductByName(productName) {
+        const productElement = $(this.getProductByName(productName));
```

Comment:
You can implement `getProductByName` so it returns an element immediately, so we don't need to wrap it with `$()` every call. Also, `await` is needed.

Reply:
fixed i guess